x/fungible/keeper/zrc20_cosmos_coins_mapping.go

```diff
@@ -0,0 +1,192 @@
+	}
+
+	// transfer from the EOA locking the assets to the owner.
+	transferred, err := k.ZRC20Transfer(ctx, zrc20ABI, zrc20Address, locker, owner, amount)
+	if err != nil {
+		return errors.Wrap(err, "failed executing transfer")
+	}
+
+	if !transferred {
+		return fmt.Errorf("transfer returned false (no success)")
+	}
+
+	return nil
+}
+
+// CheckZRC20Allowance checks if the allowance of ZRC20 tokens,
+// is equal or greater than the provided amount.
+func (k Keeper) CheckZRC20Allowance(
+	ctx sdk.Context,
+	zrc20ABI *abi.ABI,
+	owner, spender, zrc20Address common.Address,
+	amount *big.Int,
+) error {
+	if zrc20ABI == nil {
+		return fungibletypes.ErrZRC20NilABI
+	}
+
+	if amount.Sign() <= 0 || amount == nil {
+		return fungibletypes.ErrInvalidAmount
+	}
+
+	if crypto.IsEmptyAddress(owner) || crypto.IsEmptyAddress(spender) {
+		return fungibletypes.ErrZeroAddress
+	}
+
+	if err := k.IsValidZRC20(ctx, zrc20Address); err != nil {
+		return errors.Wrap(err, "ZRC20 is not valid")
+	}
+
+	allowanceValue, err := k.ZRC20Allowance(ctx, zrc20ABI, zrc20Address, owner, spender)
+	if err != nil {
+		return errors.Wrap(err, "failed while checking spender's allowance")
+	}
+
+	if allowanceValue.Cmp(amount) < 0 || allowanceValue.Cmp(big.NewInt(0)) <= 0 {
+		return fmt.Errorf("invalid allowance, got %s, wanted %s", allowanceValue.String(), amount.String())
+	}
+
+	return nil
+}
+
+// CheckZRC20Balance checks if the balance of ZRC20 tokens,
+// is equal or greater than the provided amount.
+func (k Keeper) CheckZRC20Balance(
+	ctx sdk.Context,
+	zrc20ABI *abi.ABI,
+	zrc20Address, owner common.Address,
+	amount *big.Int,
+) error {
+	if zrc20ABI == nil {
+		return fungibletypes.ErrZRC20NilABI
+	}
+
+	if amount.Sign() <= 0 || amount == nil {
+		return fungibletypes.ErrInvalidAmount
+	}
+
+	if err := k.IsValidZRC20(ctx, zrc20Address); err != nil {
+		return errors.Wrap(err, "ZRC20 is not valid")
+	}
+
+	if crypto.IsEmptyAddress(owner) {
+		return fungibletypes.ErrZeroAddress
+	}
+
+	// Check the ZRC20 balance of a given account.
+	// function balanceOf(address account)
+	balance, err := k.ZRC20BalanceOf(ctx, zrc20ABI, zrc20Address, owner)
+	if err != nil {
+		return errors.Wrap(err, "failed getting owner's ZRC20 balance")
+	}
+
+	if balance.Cmp(amount) < 0 {
+		return fmt.Errorf("invalid balance, got %s, wanted %s", balance.String(), amount.String())
+	}
+
+	return nil
+}
+
+// IsValidZRC20 returns an error whenever a ZRC20 is not whitelisted or paused.
+func (k Keeper) IsValidZRC20(ctx sdk.Context, zrc20Address common.Address) error {
+	if crypto.IsEmptyAddress(zrc20Address) {
+		return fungibletypes.ErrZRC20ZeroAddress
+	}
+
+	t, found := k.GetForeignCoins(ctx, zrc20Address.String())
+	if !found {
+		return fungibletypes.ErrZRC20NotWhiteListed
+	}
+
+	if t.Paused {
+		return fungibletypes.ErrPausedZRC20
+	}
+
+	return nil
+}
+
+// IsValidDepositAmount checks "totalSupply >= amount_to_be_locked + amount_already_locked".
```

Comment:
```suggestion
// IsValidDepositAmount checks "totalSupply >= amount_already_locked+ amount_to_be_locked ".
```
Also maybe use the the same identifiers in the comment and code

Reply:
I'm not sure if the suggested change makes any difference?

Reply:
its just a nitpick , so that the code is better aligned wth the comment